spyder/plugins/editor/widgets/editor.py

```diff
@@ -2833,15 +2833,24 @@ def _run_cell_text(self, text, editor, cell_id, debug=False):
     def dragEnterEvent(self, event):
         """Reimplement Qt method
         Inform Qt about the types of data that the widget accepts"""
+        logger.debug("DragEnterEvent:")
         source = event.mimeData()
         # The second check is necessary on Windows, where source.hasUrls()
         # can return True but source.urls() is []
         # The third check is needed since a file could be dropped from
         # compressed files. In Windows mimedata2url(source) returns None
         # Fixes spyder-ide/spyder#5218.
-        if source.hasUrls() and source.urls() and mimedata2url(source):
-            all_urls = mimedata2url(source)
+        has_urls = source.hasUrls()
+        has_text = source.hasText()
+        urls = source.urls()
+        all_urls = mimedata2url(source)
+        logger.debug("Source has_urls: {}".format(has_urls))
+        logger.debug("Source urls: {}".format(urls))
+        logger.debug("Source all_urls: {}".format(all_urls))
+        logger.debug("Source has_text: {}".format(has_text))

```

Comment:
In these lines please change `Source` to `Drag event source`, to make them easier to spot in our logs.